cli/src/cmd_auth.rs

```diff
@@ -538,8 +539,62 @@ impl CmdAuthStatus {
 
     fn error_msg(e: &oxide::Error<oxide::types::Error>) -> String {
         match e {
-            oxide::Error::ErrorResponse(ee) => format!("Error Response: {}", ee.message),
-            ee => ee.to_string(),
+            oxide::Error::CommunicationError(error) => {
+                if error.is_timeout() {
+                    "Request timed out".to_string()
+                } else if error.is_connect() {
+                    // Reqwest will just tell us there was an error; we want to
+                    // look at its internal error to understand the cause.
+                    // Regardless this is reflective of a programming error.
+                    let details = error
+                        .source()
```

Comment:
After rebasing this on top of `update-several`, we're still losing useful context here:

```
# With reqwest 0.12.8
$ oxide auth status
  Checking https://unresolvabledomainnameihope...
Profile "string" (https://unresolvabledomainnameihope) status: Failed to connect to server: client error (Connect)

# With reqwest 0.11.27
$ oxide auth status
  Checking https://unresolvabledomainnameihope...
Profile "string" (https://unresolvabledomainnameihope) status: Failed to connect to server: error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known
```